mockwebserver/src/main/java/okhttp3/internal/http2/Http2Server.java

```diff
@@ -51,31 +51,32 @@ public Http2Server(File baseDirectory, SSLSocketFactory sslSocketFactory) {
   }
 
   private void run() throws Exception {
-    ServerSocket serverSocket = new ServerSocket(8888);
-    serverSocket.setReuseAddress(true);
-
-    while (true) {
-      Socket socket = null;
-      try {
-        socket = serverSocket.accept();
-
-        SSLSocket sslSocket = doSsl(socket);
-        String protocolString = Platform.get().getSelectedProtocol(sslSocket);
-        Protocol protocol = protocolString != null ? Protocol.get(protocolString) : null;
-        if (protocol != Protocol.HTTP_2) {
-          throw new ProtocolException("Protocol " + protocol + " unsupported");
+    try (ServerSocket serverSocket = new ServerSocket(8888)) {
```

Comment:
We don’t use try-with-resources in our code ’cause it isn’t available on all of the platforms we support (Android 2.3+).